### PR TITLE
Add PipeStorage tests.

### PR DESCRIPTION
**Pull request by dekimir**
@antiagainst @AWoloszyn @dneto0 @Qining PTAL.

### CODE REVIEW

File: test/TextToBinary.PipeStorage.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2016 Google
```
Comment (antiagainst) [inline test/TextToBinary.PipeStorage.cpp]: I'm not a license lawyer. So I'd think @dneto0 should check and make sure this is correct.

File: test/TextToBinary.PipeStorage.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+// Assembler tests for instructions in the "Barrier Instructions" section
```
Comment (antiagainst) [inline test/TextToBinary.PipeStorage.cpp]: I think the following cases are testing instructions from "[Pipe Instructions](https://www.khronos.org/registry/spir-v/specs/1.1/SPIRV.html#_a_id_pipe_a_pipe_instructions)".

**Comment by antiagainst** [comment]
-1. Please fix the comment. :)

File: test/TextToBinary.PipeStorage.cpp (hunk 1 above)
Comment (dekimir) [inline test/TextToBinary.PipeStorage.cpp]:
```
test/ValidationState.cpp:// Copyright (c) 2016 Google
test/TextToBinary.PipeStorage.cpp:// Copyright (c) 2016 Google
test/SoftwareVersion.cpp:// Copyright (c) 2015-2016 Google Inc.
test/TextToBinary.SubgroupDispatch.cpp:// Copyright (c) 2016 Google
source/software_version.cpp:// Copyright (c) 2015-2016 Google Inc.
```

File: test/TextToBinary.PipeStorage.cpp (hunk 2 above)
Comment (dekimir) [inline test/TextToBinary.PipeStorage.cpp]: Removed the comment.

**Comment by antiagainst** [comment]
Code looks good to me. Just the license now.

File: test/TextToBinary.PipeStorage.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  EXPECT_THAT(CompileFailure("%1 = OpConstantPipeStorage %2 3 4 5 %6 %7",
+                             SPV_ENV_UNIVERSAL_1_1),
+              Eq("'=' expected after result id."));
+}
+
+TEST_F(OpConstantPipeStorageTest, ArgumentTypes) {
+  EXPECT_THAT(CompileFailure("%1 = OpConstantPipeStorage %2 %3 4 5",
+                             SPV_ENV_UNIVERSAL_1_1),
+              Eq("Invalid unsigned integer literal: %3"));
+  EXPECT_THAT(CompileFailure("%1 = OpConstantPipeStorage %2 %3 4 5",
```
Comment (dneto0) [inline test/TextToBinary.PipeStorage.cpp]: This looks identical to the previous check.

File: test/Validate.Capability.cpp [hunk 4]
```diff
@@ -127,7 +127,9 @@ const vector<string>& AllCapabilities() {
     "StorageImageReadWithoutFormat",
     "StorageImageWriteWithoutFormat",
     "MultiViewport",
-    "SubgroupDispatch"};
+    "SubgroupDispatch",
+    "NamedBarrier",
```
Comment (dneto0) [inline test/Validate.Capability.cpp]: NamedBarrier is off-topic for this CL.   Ok to have in this CL if you update the commit message to mention it as a drive-by addition.

Also, I suspect you'll want another CL for validator tests of the PipeStorage capability.

**Comment by dneto0** [comment]
Copryright is ok.
-1 just for the duplicated check in one of the tests.

File: test/Validate.Capability.cpp (hunk 4 above)
Comment (dekimir) [inline test/Validate.Capability.cpp]: Turns out that was already done on master via other PRs.  This diff goes away after rebasing.

**Comment by dekimir** [comment]
Fixed the duplicate check, rebased, and squashed.

**Comment by dneto0** [comment]
+2
